server/src/main/java/org/elasticsearch/index/translog/TruncateTranslogAction.java

```diff
@@ -177,11 +177,19 @@ private boolean isTranslogClean(ShardPath shardPath, String translogUUID) throws
             final TranslogConfig translogConfig = new TranslogConfig(shardPath.getShardId(), translogPath,
                 indexSettings, BigArrays.NON_RECYCLING_INSTANCE);
             long primaryTerm = indexSettings.getIndexMetaData().primaryTerm(shardPath.getShardId().id());
-            final TranslogDeletionPolicy translogDeletionPolicy =
-                new TranslogDeletionPolicy(indexSettings.getTranslogRetentionSize().getBytes(),
-                    indexSettings.getTranslogRetentionAge().getMillis());
+            // We open translog to check for corruption, do not clean anything.
```

Comment:
I made this change because we do not set `minTranslogGenerationForRecovery` here, then we trip an [assertion](https://github.com/elastic/elasticsearch/blob/c251450a2fec134084992a0ed4a288bada1e8c3e/server/src/main/java/org/elasticsearch/index/translog/Translog.java#L1672-L1674) when releasing a snapshot. The test `RemoveCorruptedShardDataCommandIT#testCorruptIndex` would fail if we disable the translog retention.